### PR TITLE
Multiple runtimes [v3-v4]

### PR DESCRIPTION
**Pull request by raynaudoe**
- Fix LookUpSource type definition and routing for V3 and V4 in parser.c
- Fix keys type
- Updated testcases for V3 and V4.
- Bump major version

### CODE REVIEW

File: app/Makefile.version [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # This is the `transaction_version` field of `Runtime`
-APPVERSION_M=3
+APPVERSION_M=4
 # This is the `spec_version` field of `Runtime`
 APPVERSION_N=26
```
Comment (jleni) [inline app/Makefile.version:4]:
```suggestion
APPVERSION_N=28
```

File: app/Makefile.version [hunk 2]
```diff
@@ -1,5 +1,5 @@
...
 APPVERSION_P=1
```
Comment (jleni) [inline app/Makefile.version:5]:
```suggestion
APPVERSION_P=0
```

File: app/src/parser_impl.h [hunk 3]
```diff
@@ -94,6 +94,17 @@ GEN_DEC_READFIX_UNSIGNED(64);
...
+    pd_##TYPE##_V##VERSION##_t dummy;                                            \
+    compactInt_t clen;                                              \
+    CHECK_PARSER_ERR(_readCompactInt(c, &clen));                    \
+    CHECK_PARSER_ERR(_getValue(&clen, &v->_len));                   \
+    v->_ptr = c->buffer + c->offset;                                \
+    v->_lenBuffer = c->offset;                                      \
+    for (uint64_t i = 0; i < v->_len; i++ ) CHECK_ERROR(_read##TYPE##_V##VERSION (c, &dummy));  \
+    v->_lenBuffer = c->offset - v->_lenBuffer;                      \
+    return parser_ok;
+
```
Comment (jleni) [inline app/src/parser_impl.h:107]: :sunglasses:

File: app/src/substrate_types.c [hunk 4]
```diff
@@ -112,6 +128,20 @@ parser_error_t _readBalanceOf(parser_context_t* c, pd_BalanceOf_t* v)
     return _readBalance(c, &v->value);
 }
 
+parser_error_t _readBytes(parser_context_t* c, pd_Bytes_t* v)
+{
+
```
Comment (jleni) [inline app/src/substrate_types.c:133]:
```suggestion
```

File: app/src/substrate_types.c [hunk 5]
```diff
@@ -126,28 +156,42 @@ parser_error_t _readu8_array_20(parser_context_t* c, pd_u8_array_20_t* v){
     GEN_DEF_READARRAY(20)
 }
 
+parser_error_t _readCall(parser_context_t* c, pd_Call_t* v)
+{
+
```
Comment (jleni) [inline app/src/substrate_types.c:161]:
```suggestion
```

File: app/src/substrate_types.c [hunk 6]
```diff
@@ -126,28 +156,42 @@ parser_error_t _readu8_array_20(parser_context_t* c, pd_u8_array_20_t* v){
...
 parser_error_t _readHeader(parser_context_t* c, pd_Header_t* v)
 {
 
     return parser_not_supported;
 }
 
-parser_error_t _readLookupSource(parser_context_t* c, pd_LookupSource_t* v){
+parser_error_t _readProposal(parser_context_t* c, pd_Proposal_t* v)
+{
```
Comment (jleni) [inline app/src/substrate_types.c:176]:
```suggestion
```

File: app/src/substrate_types_V3.c [hunk 7]
```diff
@@ -226,14 +220,22 @@ parser_error_t _readKeys_V3(parser_context_t* c, pd_Keys_V3_t* v){
     GEN_DEF_READARRAY(5 * 32)
 }
 
+parser_error_t _readLookupSource_V3(parser_context_t* c, pd_LookupSource_V3_t* v){
+
```
Comment (jleni) [inline app/src/substrate_types_V3.c:224]:
```suggestion
```

**Review by jleni — CHANGES_REQUESTED** [review]

File: app/Makefile.version (hunk 1 above)
Comment (jleni) [reply app/Makefile.version:4]: Actually I am not sure about this value

Comment (jleni) [reply app/Makefile.version:4]: yes, it should be 28
https://github.com/paritytech/polkadot/releases/tag/v0.8.28

File: app/src/substrate_types.c (hunk 4 above)
Comment (jleni) [reply app/src/substrate_types.c:133]: odd spaces here

**Comment by jleni** [comment]
Awesome work! :cocktail:

**Comment by jleni** [comment]
:bird:

File: app/src/substrate_types.c (hunk 4 above)
Comment (raynaudoe) [reply app/src/substrate_types.c:133]: yeah.. need to tune up the linter a little bit more

**Comment by raynaudoe** [comment]
Thanks!
all suggested commits are applied

**Pull request merged**